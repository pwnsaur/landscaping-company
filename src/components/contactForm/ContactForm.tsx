import { useState } from 'react';
import styled from 'styled-components';
import { FormData } from '@/types/contentfulTypes';
import Textarea from './Textarea';
import Input from './Input';
import Button from './Button';

type ContactFormProps = {
  onSubmit: (formData: FormData) => void;
};

const ContactForm = ({ onSubmit }: ContactFormProps) => {
  const [name, setName] = useState<string>('');
  const [email, setEmail] = useState<string>('');
  const [phone, setPhone] = useState<string>('');
  const [message, setMessage] = useState<string>('');

  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const formData = {
      name,
      email,
      phone,
      message,
    };

    try {
      console.log(formData);
      onSubmit(formData);
      setName('');
      setEmail('');
      setPhone('');
      setMessage('');
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <Form onSubmit={handleSubmit}>
      <Input
        type='text'
        onChange={(e) => setName(e.target.value)}
        value={name}
        placeholder='Vārds, uzvārds'
        required
      />
      <Input
        type='email'
        onChange={(e) => setEmail(e.target.value)}
        value={email}
        placeholder='Epasts'
        required
      />
      <Input
        type='tel'
        onChange={(e) => setPhone(e.target.value)}
        value={phone}
        placeholder='Tālrunis'
        required
      />
      <Textarea
        onChange={(e) => setMessage(e.target.value)}
        value={message}
        placeholder='Ievadiet ziņojumu'
        required
      />
      <Button type='submit'>Apstiprināt</Button>
    </Form>
  );
};

export default ContactForm;

const Form = styled.form`
  width: 100%;
  /* max-width: 500px; */
  margin: 2rem auto 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;
`;
