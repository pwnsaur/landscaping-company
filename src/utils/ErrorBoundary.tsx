import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
