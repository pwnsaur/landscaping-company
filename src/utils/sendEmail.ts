import axios, { AxiosResponse } from 'axios';

import { FormData } from '@/types/contactForm';

type ServerResponse = {
  messageId: string;
  response: string;
};

type SendEmailFn = (
  formData: FormData
) => Promise<AxiosResponse<ServerResponse>>;

const sendEmail: SendEmailFn = async (formData: FormData) => {
  return axios({
    method: 'post',
    url: '/api/send-mail',
    data: formData,
  });
};

export default sendEmail;
