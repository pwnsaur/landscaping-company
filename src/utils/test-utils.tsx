import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
