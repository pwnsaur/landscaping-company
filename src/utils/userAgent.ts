export function isMobileUserAgent(userAgent: string): boolean {
  return /Android|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
}
