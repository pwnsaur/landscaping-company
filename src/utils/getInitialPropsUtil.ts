import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
