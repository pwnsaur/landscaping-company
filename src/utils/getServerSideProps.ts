import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
