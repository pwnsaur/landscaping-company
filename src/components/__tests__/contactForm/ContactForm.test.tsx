import ContactForm from '@/components/contactForm/ContactForm';
import { render } from '@/utils/test-utils';

describe('ContactForm component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(<ContactForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
