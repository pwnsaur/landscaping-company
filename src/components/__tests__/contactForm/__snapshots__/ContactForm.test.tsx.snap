// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component matches the snapshot 1`] = `
<DocumentFragment>
  .c3 {
  padding: 0.5rem;
  min-width: 5rem;
  background-color: #21412a;
  color: #fff;
  border: none;
  font-size: 1rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  cursor: pointer;
}

.c3:hover {
  background-color: #05ac05;
}

.c3:disabled {
  color: #696767;
  cursor: not-allowed;
}

.c1 {
  padding: 10px;
  border: 1px solid #ccc;
  font-size: 1rem;
  margin: 15px 0;
  width: 100%;
  outline: none;
}

.c1:focus {
  border-color: #05ac05;
}

.c1::-webkit-input-placeholder {
  color: #aaa;
}

.c1::-moz-placeholder {
  color: #aaa;
}

.c1:-ms-input-placeholder {
  color: #aaa;
}

.c1::placeholder {
  color: #aaa;
}

.c2 {
  padding: 1rem;
  border: 1px solid #ccc;
  font-size: 1rem;
  margin: 1.5rem 0 2.5rem;
  width: 100%;
  height: 10rem;
  resize: none;
  outline: none;
}

.c2:focus {
  border-color: #05ac05;
}

.c2::-webkit-input-placeholder {
  color: #aaa;
}

.c2::-moz-placeholder {
  color: #aaa;
}

.c2:-ms-input-placeholder {
  color: #aaa;
}

.c2::placeholder {
  color: #aaa;
}

.c0 {
  width: 100%;
  margin: 2rem auto 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<form
    class="c0"
  >
    <input
      class="c1"
      id="name"
      placeholder="Vārds, uzvārds"
      required=""
      type="text"
      value=""
    />
    <input
      class="c1"
      id="email"
      placeholder="Epasts"
      required=""
      type="email"
      value=""
    />
    <input
      class="c1"
      id="phone"
      placeholder="Tālrunis"
      required=""
      type="tel"
      value=""
    />
    <textarea
      class="c2"
      id="message"
      placeholder="Ievadiet ziņojumu"
      required=""
    />
    <button
      class="c3"
    >
      Sūtīt
    </button>
  </form>
  .c0 {
  display: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  height: 3rem;
  background-color: #ff3333;
  color: #333;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  z-index: 5;
}

.c1 {
  font-size: 1rem;
  font-weight: bold;
  margin: 0;
  padding: 0;
}

<div
    class="c0"
  >
    <p
      class="c1"
    />
  </div>
</DocumentFragment>
`;
